application/controllers/admin/export.php

```diff
@@ -156,6 +155,9 @@ public function exportresults()
         $surveybaselang = $survey->language;
         $exportoutput = "";
 
+        // Avoid randomization of the fieldmap
+        unset($_SESSION['survey_'.$iSurveyID]['fieldmap-'.$iSurveyID.'-randMaster']);
```

Comment:
I really think we need a better way than unset a session var …

Reply:
For example check a param or some LimeExpressionManager::State somewher ( sPreviewMode ? for example : we use it for logic for example : logic/export ?)
here
https://github.com/LimeSurvey/LimeSurvey/blob/91375a58e22334af287e652885599142954a676f/application/helpers/common_helper.php#L1897

Reply:
I didn't want to update the createFieldMap function.
Still, could be possible.

Maybe add a new optional param?

Reply:
I don't know, maybe you're right.

Why not just checking with `LimeExpressionManager::SetDirtyFlag();` or `killSurveySession($surveyid);` ?

Because : it work in 2.6lts without new params and without hacking  `$_SESSION` directly.

Reply:
That's a good idea as well. Will try it